x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/downsample/FieldValueFetcher.java

```diff
@@ -8,31 +8,59 @@
 package org.elasticsearch.xpack.downsample;
 
 import org.apache.lucene.index.LeafReaderContext;
-import org.elasticsearch.index.fielddata.FormattedDocValues;
 import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.fielddata.LeafFieldData;
 import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.index.mapper.NumberFieldMapper;
 import org.elasticsearch.index.query.SearchExecutionContext;
-import org.elasticsearch.search.DocValueFormat;
 import org.elasticsearch.xpack.aggregatemetric.mapper.AggregateDoubleMetricFieldMapper;
 
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Locale;
 
 /**
  * Utility class used for fetching field values by reading field data
  */
 class FieldValueFetcher {
 
-    private final MappedFieldType fieldType;
-    private final IndexFieldData<?> fieldData;
+    protected final MappedFieldType fieldType;
+    protected final IndexFieldData<?> fieldData;
     private final AbstractRollupFieldProducer rollupFieldProducer;
 
+    private static class RollupFieldProducersFactory {
+
+        static AbstractRollupFieldProducer createProducer(final MappedFieldType fieldType, final String name) {
```

Comment:
Can we add unit tests for this method? I don't see unit tests for this method now or for the original `createRollupFieldProducer()` method.

Reply:
Ok.